vm/src/obj/objrange.rs

```diff
@@ -402,6 +404,25 @@ impl PyRange {
         }
     }
 
+    #[pymethod(name = "__hash__")]
+    fn hash(zelf: PyRef<Self>, vm: &VirtualMachine) -> PyResult<pyhash::PyHash> {
+        let length = zelf.length();
+        let len = length.as_bigint().clone();
+        let a = length.into_ref(vm).into_object();
+        let mut b = vm.get_none();
+        let mut c = vm.get_none();
+        if !len.is_zero() {
+            b = zelf.start.clone().into_object();
+            if !len.is_one() {
+                c = zelf.step(vm).into_object();
+            }
+        };
+        let tuple = PyTuple {
+            elements: vec![a, b, c],
+        };
```

Comment:
I'm not sure if PyTuple creation is the way above.
Is there a better way?

Reply:
looking for `vm.ctx.new_tuple`?

Reply:
It looks like PyTuple's hash just uses `pyhash::hash_iter`, so you could probably do the same.